tests/commons/src/main/java/org/codice/ddf/admin/comp/test/AbstractComponentTest.java

```diff
@@ -0,0 +1,177 @@
+import static org.ops4j.pax.exam.karaf.options.KarafDistributionOption.karafDistributionConfiguration;
+import static org.ops4j.pax.exam.karaf.options.KarafDistributionOption.keepRuntimeFolder;
+import static org.ops4j.pax.exam.karaf.options.KarafDistributionOption.logLevel;
+import static junit.framework.TestCase.fail;
+
+import java.io.File;
+import java.lang.reflect.Proxy;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.inject.Inject;
+
+import org.apache.karaf.features.BootFinished;
+import org.codice.ddf.itests.common.AdminConfig;
+import org.codice.ddf.itests.common.ServiceManager;
+import org.codice.ddf.itests.common.ServiceManagerImpl;
+import org.codice.ddf.itests.common.ServiceManagerProxy;
+import org.codice.ddf.itests.common.annotations.BeforeExam;
+import org.codice.ddf.itests.common.annotations.PaxExamRule;
+import org.codice.ddf.itests.common.security.SecurityPolicyConfigurator;
+import org.junit.Rule;
+import org.ops4j.pax.exam.Configuration;
+import org.ops4j.pax.exam.Option;
+import org.ops4j.pax.exam.karaf.options.KarafDistributionOption;
+import org.ops4j.pax.exam.karaf.options.LogLevelOption;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.osgi.service.metatype.MetaTypeService;
+
+public abstract class AbstractComponentTest {
+
+    @Rule
+    public PaxExamRule paxExamRule = new PaxExamRule(this);
+
+    /**
+     * To make sure the tests run only when the boot features are fully installed
+     */
+    @Inject
+    BootFinished bootFinished;
+
+    @Inject
+    protected ConfigurationAdmin configurationAdmin;
+
+    @Inject
+    protected MetaTypeService metatypeService;
+
+    protected AdminConfig adminConfig;
+
+    protected ServiceManager serviceManager;
+
+    protected SecurityPolicyConfigurator securityPolicyConfigurator;
+
+    /**
+     * Adds the given FeatureFile to the feature-repo. If the FeatureFile is an instance of a Feature,
+     * it will be started when the test begins.
+     *
+     * @return
+     */
+    public abstract List<Feature> features();
+
+    @Configuration
+    public Option[] config() throws Exception {
+        List<Option> features = features().stream()
+                .map(feature -> {
+                    if (feature.isBootFeature()) {
+                        return KarafDistributionOption.features(feature.getUrl(), feature.getFeatureName());
+                    } else {
+                        return KarafDistributionOption.features(feature.getUrl());
+                    }
+                })
+                .collect(Collectors.toList());
+
+        return Stream.of(distributionSettings(), features, configurableSettings(), configureVmOptions(), customSettings())
+                .flatMap(Collection::stream)
+                .toArray(Option[]::new);
+    }
+
+    public List<Option> customSettings() {
```

Comment:
❓ These other `*settings()` methods other than `config()`, it looks like they could be `protected` instead of `public`. Am I missing something?